org.eclipse.winery.generators.ia/src/test/java/org/eclipse/winery/generators/ia/GeneratorTest.java

```diff
@@ -13,126 +13,96 @@
-        op1ip2.setType("xs:string");
-        op1InputParameters.getInputParameter().add(op1ip2);
-        opIn.setInputParameters(op1InputParameters);
-
-        TOperation opOut = new TOperation();
-        opOut.setName("OpOut");
-        tinterface.getOperation().add(opOut);
-
-        TOperation.OutputParameters op1OutputParameters = new TOperation.OutputParameters();
-        TParameter op1op1 = new TParameter();
-        op1op1.setName("op1op1");
-        op1op1.setType("xs:string");
-        op1OutputParameters.getOutputParameter().add(op1op1);
-        TParameter op1op2 = new TParameter();
-        op1op2.setName("op1op2");
-        op1op1.setType("xs:string");
-        op1OutputParameters.getOutputParameter().add(op1op2);
-        opOut.setOutputParameters(op1OutputParameters);
-
-        TNodeType nodeType = new TNodeType();
-        nodeType.setName("test");
-        nodeType.setTargetNamespace("http://asd.com");
-
-        Generator gen = new Generator(tinterface, "org.opentosca.ia", new URL("http://asd.com"), "testname", WORKING_DIR.toFile());
-        Path generateProject = gen.generateProject();
-        System.out.println(generateProject);
+    public void testMultipleOpsWithInOutParams() throws Exception {
+        TInterface i = new TInterface();
+        i.setName("http://www.example.org/interfaces/lifecycle");
+
+        TOperation op;
+        TOperation.InputParameters input;
+        TOperation.OutputParameters output;
+        TParameter param;
+
+        op = new TOperation();
+        op.setName("install");
+        i.getOperation().add(op);
+
+        input = new TOperation.InputParameters();
+        param = new TParameter();
+        param.setName("VMIP");
+        param.setType("xs:string");
+        input.getInputParameter().add(param);
+        param = new TParameter();
+        param.setName("DBMSUsername");
+        param.setType("xs:string");
+        input.getInputParameter().add(param);
+        op.setInputParameters(input);
+
+        output = new TOperation.OutputParameters();
+        param = new TParameter();
+        param.setName("Output");
+        param.setType("xs:string");
+        output.getOutputParameter().add(param);
+        op.setOutputParameters(output);
+
+        op = new TOperation();
+        op.setName("uninstall");
+        i.getOperation().add(op);
+
+        input = new TOperation.InputParameters();
+        param = new TParameter();
+        param.setName("SomeLongParameterName");
+        param.setType("xs:string");
+        input.getInputParameter().add(param);
+        param = new TParameter();
+        param.setName("Port");
+        param.setType("xs:string");
+        input.getInputParameter().add(param);
+        op.setInputParameters(input);
+
+        op.setOutputParameters(output);
+
+        Generator gen = new Generator(i, "org.opentosca.ia.test", new URL("http://test.com"), "TestMultipleOpsWithInOutParams", wd.toFile());
+        gen.generateProject();
```

Comment:
Why don't you compare your expected file against the generated one?

Reply:
The test didn't have assertions in the first place. It is just used for quick testing (and manual checking if the project has been generated successfully). So, I don't want to extend this here... I just put two test methods together and made minor changes, e.g. to identify the test output properly.